src/test/java/teammates/test/cases/datatransfer/AccountAttributesTest.java

```diff
@@ -133,6 +133,65 @@ public void testLegacyAccountEntityToAttributes() {
 
     }
 
+    @Test
+    public void getCopy_shouldCreateDeepCopy() {
+        String accountGoogleId = "johnDoe";
+
+        AccountAttributes account = AccountAttributes.builder()
+                .withGoogleId(accountGoogleId)
+                .withName("John")
+                .withEmail("johnDoeEmail")
+                .withInstitute("johnson")
+                .withIsInstructor(true)
+                .build();
+        account.isInstructor = true;
+
+        account.studentProfile = StudentProfileAttributes.builder()
+                .withGoogleId(accountGoogleId)
+                .withShortName("John")
+                .withEmail("person@email.com")
+                .withInstitute("profile institute test profile")
+                .withNationality("nation")
+                .withGender("male")
+                .withMoreInfo("some test info 221")
+                .withPictureKey("picture key")
+                .build();
+
+        AccountAttributes copy = account.getCopy();
+
+        assertFalse(account.equals(copy));
+        assertTrue(account.isInstructor);
+        assertEquals(account.googleId, copy.googleId);
+        assertEquals(account.name, copy.name);
+        assertEquals(account.institute, copy.institute);
+        assertEquals(account.email, copy.email);
+
+        assertFalse(account.studentProfile.equals(copy.studentProfile));
+    }
+
+    @Test
+    public void getCopy_shouldCreateDeepCopyIfAllFieldsNull() {
+        AccountAttributes account = AccountAttributes.builder()
+                .withGoogleId(null)
+                .withName(null)
+                .withEmail(null)
+                .withInstitute(null)
+                .withIsInstructor(false)
+                .withStudentProfileAttributes(null)
+                .build();
+
+        AccountAttributes copy = account.getCopy();
+
+        assertFalse(account.equals(copy));
+        assertFalse(account.isInstructor);
+
+        assertNull(copy.studentProfile);
+        assertNull(account.googleId, copy.googleId);
```

Comment:
The second parameter of `assertNull` is the message. (Same problem for the code below)
```java
assertNull(account.googleId); // this is enough as we just want to validate whether the googleId is null or not

assertNull(account.googleId, "googleId should be null"); // or you can provide failure message
```
Check more here.
http://javadox.com/org.testng/testng/6.8.5/org/testng/asserts/Assertion.html#assertNull(java.lang.Object)